sources/packages/api/src/services/education-program/education-program.service.ts

```diff
@@ -274,4 +431,58 @@ export class EducationProgramService extends RecordDataModelService<EducationPro
       .orderBy("programs.name")
       .getMany();
   }
+  /**
+   * Gets program details with program id.
+   * @param programId Program id.
+   * @returns program
+   */
+  async getEducationProgramDetails(
+    programId: number,
+  ): Promise<EducationProgram> {
+    return this.repo
+      .createQueryBuilder("programs")
+      .select([
+        "programs.id",
+        "programs.name",
+        "programs.description",
+        "programs.credentialType",
+        "programs.cipCode",
+        "programs.nocCode",
+        "programs.sabcCode",
+        "programs.approvalStatus",
+        "programs.programIntensity",
+        "programs.institutionProgramCode",
+        "programs.regulatoryBody",
+        "programs.deliveredOnSite",
+        "programs.deliveredOnline",
+        "programs.deliveredOnlineAlsoOnsite",
+        "programs.sameOnlineCreditsEarned",
+        "programs.earnAcademicCreditsOtherInstitution",
+        "programs.courseLoadCalculation",
+        "programs.completionYears",
+        "programs.eslEligibility",
+        "programs.hasJointInstitution",
+        "programs.hasJointDesignatedInstitution",
+        "programs.institutionProgramCode",
+        "programs.minHoursWeek",
+        "programs.isAviationProgram",
+        "programs.minHoursWeekAvi",
+        "programs.hasMinimumAge",
+        "programs.minHighSchool",
+        "programs.requirementsByInstitution",
+        "programs.requirementsByBCITA",
+        "programs.hasWILComponent",
+        "programs.isWILApproved",
+        "programs.wilProgramEligibility",
+        "programs.hasTravel",
+        "programs.travelProgramEligibility",
+        "programs.hasIntlExchange",
+        "programs.intlExchangeProgramEligibility",
+        "programs.programDeclaration",
+        "institution.id",
+      ])
+      .innerJoin("programs.institution", "institution")
```

Comment:
If the _institutionId_ is all that we need maybe we can avoid adding the _innerJoin_ if we add a @RelationId to _institutionId_ in _EducationProgram_ entity model.

Reply:
as discussed, I was not able to access the `institution id` by specifying the name 
![image](https://user-images.githubusercontent.com/77353155/152405556-fa568761-f41b-4968-9387-2e652d43551b.png)
it is working only when we select whole entity like 
![image](https://user-images.githubusercontent.com/77353155/152405702-1b990a8b-2735-492f-91a4-5ad526cbf77b.png)